packages/gatsby-source-mongodb/src/gatsby-node.js

```diff
@@ -21,12 +21,15 @@ exports.sourceNodes = (
   let connectionExtraParams = getConnectionExtraParams(
     pluginOptions.extraParams
   )
-  const connectionURL = `mongodb://${authUrlPart}${serverOptions.address}:${
-    serverOptions.port
-  }/${dbName}${connectionExtraParams}`
+  const connectionURL = pluginOptions.connectionString
+    ? `${pluginOptions.connectionString}/${dbName}${connectionExtraParams}`
+    : `mongodb://${authUrlPart}${serverOptions.address}:${
+        serverOptions.port
+      }/${dbName}${connectionExtraParams}`
 
   return MongoClient.connect(connectionURL)
-    .then(db => {
+    .then(client => {
+      let db = client.db(dbName)
```

Comment:
nit: but if this is not re-assigned, I generally use a `const`

```suggestion
      const db = client.db(dbName)
```

Reply:
Awesome :) I've added a `status: blocked` label to this so that we can go through that session before getting this merged and validated.

Thanks for doing that!